src/Globals.h

```diff
@@ -35,11 +35,15 @@
 		// The CRT has a definition for this operator new that stores the debugging info for leak-finding later.
 	#endif
 
+	#define UNREACHABLE_INTRINSIC __assume(false)
+
 #elif defined(__GNUC__)
 
 	// TODO: Can GCC explicitly mark classes as abstract (no instances can be created)?
 	#define abstract
 
+	#define UNREACHABLE_INTRINSIC __builtin_unreachable()
+
```

Comment:
This will check the paths at compile time - right?

Reply:
Ok - it's more complicated. But It tells the compiler/optimiser that there is no more code that can be executed